client/asset/eth/eth.go

```diff
@@ -3691,6 +3749,59 @@ func (w *assetWallet) checkFindRedemptions() {
 	}
 }
 
+// sumPendingTxs sums the expected incoming and outgoing values in unconfirmed
+// transactions stored in pendingTxs. Not used if the node is a mempoolFetcher.
+func (w *assetWallet) sumPendingTxs() (out, in uint64) {
+	w.tipMtx.RLock()
+	tip := w.currentTip.Number.Uint64()
+	w.tipMtx.RUnlock()
+
+	isETH := w.assetID == BipID
+	addPendingSend := func(pt *pendingTx) {
+		in += pt.in
+		if isETH {
+			if pt.assetID == BipID {
+				out += pt.out + pt.fees
+			} else {
+				out += pt.fees
+			}
+			return
+		}
+		// token
+		out += pt.out
+	}
+
+	w.pendingTxMtx.Lock()
+	defer w.pendingTxMtx.Unlock()
+	for txHash, pt := range w.pendingTxs {
+		if !isETH && pt.assetID != w.assetID {
+			continue
+		}
+		if pt.lastCheck == tip {
+			// Expect nothing has changed since our last check.
+			addPendingSend(pt)
+			continue
+		}
+		confs, err := w.node.transactionConfirmations(w.ctx, txHash)
```

Comment:
I had an idea to use fetching the receipt here. That way we can leverage the receipt cache, possibly avoiding some calls. I also added the tx to the receipt cache so it does not need to fetch the tx every time before checking the cache.

Fetching the receipt and checking that error also works because the tx must be mined to get the receipt.

Also, if there are multiple mempool tx to check, and they all use the default timeout and for some reason time out, this is blocking and could take a while.

Reply:
If there are multiple `pendingTx`, the checks could be parallelized.